### DISCUSSION TITLE
worker service selector selects pods launched by Kubernetes executor with default charts pod template

### DISCUSSION THREAD

**Discussion by pewpewp3w** [apache/airflow#25233] [General]
### Official Helm Chart version

1.6.0 (latest released)

### Apache Airflow version

2.3.3 (latest released)

### Kubernetes Version

does not matter

### Helm Chart configuration

important part is
`executor: CeleryKubernetesExecutor`

### What happened

[This](https://github.com/apache/airflow/blob/helm-chart/1.6.0/chart/templates/workers/worker-service.yaml#L37) selector [matches](https://github.com/apache/airflow/blob/helm-chart/1.6.0/chart/files/pod-template-file.kubernetes-helm-yaml#L27) pods, launched from default pod template.

Having a high amount of Kubernetes executor jobs, it is possible to obtain a continuously updating list of service endpoints. In some cluster configurations, this leads to unexpected increased load to other cluster workloads. For example, istio with default configuration will constantly produce a high amount of xds pushes to all istio-enabled pods.

### What you think should happen instead

Since I don't see the point in adding KubernetesExecutor pods endpoints to worker service, I propose to change the default pod labels in pod-template-file.kubernetes-helm-yaml

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
I am not sure what you proposes. Maybe open PR, this will be easier to discuss (changing that into discussion, as this is not a bug.  - note that you do not need to have issue opened in order to create PR).
